docs/protocol/core/v3.0.rst

```diff
@@ -728,17 +728,221 @@ For example, the JSON document below defines a group::
     }
 
 
+Stores
+------
+
+A Zarr store is a system that can be used to store and retrieve data
+from a Zarr hierarchy. For a store to be compatible with this
+protocol, it must support a set of operations defined in the `Store
+interface`_ subsection. The store interface can be implemented using a
+variety of underlying storage technologies, described in the
+subsection on `Store implementations`_.
+
 Store interface
----------------
+~~~~~~~~~~~~~~~
+
+The store interface is intended to be simple to implement using a
+variety of different underlying storage technologies. It is defined in
+a general way here, but it should be straightforward to translate into
+a software interface in any given programming language. The goal is
+that an implementation of this specification could be modular and
+allow for different store implementations to be used.
+
+The store interface defines a set of operations involving `keys` and
+`values`. In the context of this interface, a `key` is any ASCII
+string containing only characters in the ranges ``a-z``, ``A-Z``,
+``0-9``, or in the set ``/.-_`` and a `value` is any sequence of
+bytes. It is assumed that the store holds (`key`, `value`) pairs, with
+only one such pair for any given `key`. I.e., a store is a mapping
+from keys to values.
+
+The following operations are **required**:
+
+``get`` - Retrieve the `value` associated with a given `key`.
+
+    | Parameters: `key`
+    | Output: `value`
+
+``set`` - Store a (`key`, `value`) pair.
+
+    | Parameters: `key`, `value`
+    | Output: none
+
+``delete`` - Delete the given key/value pair from the store.
+
+    | Parameters: `key`
+    | Output: none
+
+The following operations are **optional**, but if supported may be
+used by an implementation of this protocol to provide additional
+functionality and/or optimisations:
+
+``list`` - Retrieve all `keys` in the store.
 
-TODO define the store interface
+    | Parameters: none
+    | Output: set of `keys`
+
+``listpre`` - Retrieve all keys with a given prefix.
```

Comment:
I don't think any of my filesystems map layers have this, but clearly they can and should